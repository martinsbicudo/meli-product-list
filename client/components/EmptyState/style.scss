.EmptyState {
  width: 100%;
  padding: 2rem 0;

  &Figure {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 10rem;
    height: 10rem;
    border-radius: 50%;
    background-color: map-get($colors, "meli-gallery");
    margin: 0 auto;
    margin-bottom: 2rem;

    & > img,
    & > svg { width: 5rem; }
  }

  &Title,
  &Description { text-align: center; }

  &Title {
    font-size: 2.2rem;
    margin-bottom: .5rem;
  }

  &Description {
    font-size: 1.4rem;

    &:not(:last-child) { margin-bottom: 2rem; }
  }
}
