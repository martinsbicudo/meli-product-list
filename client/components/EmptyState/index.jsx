import React from 'react';
import { string, node, oneOfType } from 'prop-types';
import classnames from 'classnames';
import style from './style.scss';

const EmptyState = ({
  className,
  image,
  title,
  description,
  children,
}) => (
  <div
    className={classnames(
      style.EmptyState,
      className,
    )}
  >
    <figure className={style.EmptyStateFigure}>{image}</figure>
    <h1 className={style.EmptyStateTitle}>{title}</h1>
    <p className={style.EmptyStateDescription}>{description}</p>
    {children}
  </div>
);

EmptyState.propTypes = {
  className: string,
  image: node.isRequired,
  title: oneOfType([string, node]).isRequired,
  description: oneOfType([string, node]),
  children: oneOfType([string, node]),
};

EmptyState.defaultProps = {
  className: '',
  description: '',
  children: null,
};

export default EmptyState;
