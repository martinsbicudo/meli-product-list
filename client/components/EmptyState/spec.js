import EmptyState from '.';

describe('Components - <EmptyState />', () => {
  const testMessageImage = 'TEST_IMAGE';
  const testMessageTitle = 'TEST_TITLE';
  const initialState = {
    image: <img alt={testMessageImage} />,
    title: testMessageTitle,
  };
  const createWrapper = (
    props = {},
    children = null,
  ) => render(
    <EmptyState
      {...{
        ...initialState,
        ...props,
      }}
    >
      <span>{children}</span>
    </EmptyState>,
  );

  test('Should finds Image and Title', async () => {
    const { getByText, getByAltText } = createWrapper();

    expect(getByAltText(testMessageImage)).toBeInTheDocument();
    expect(getByText(testMessageTitle)).toBeInTheDocument();
  });

  test('Should finds Children', async () => {
    const testMessageChildren = 'TEST_CHILDREN';
    const { getByText } = createWrapper(null, <span>{testMessageChildren}</span>);

    expect(getByText(testMessageChildren)).toBeInTheDocument();
  });

  test('Should finds Description', async () => {
    const testMessageDescription = 'TEST_CHILDREN';
    const { getByText } = createWrapper({
      description: <span>{testMessageDescription}</span>,
    });

    expect(getByText(testMessageDescription)).toBeInTheDocument();
  });

  test('Wrapper Snapshot', () => {
    const { asFragment } = createWrapper();

    expect(asFragment()).toMatchSnapshot();
  });
});
