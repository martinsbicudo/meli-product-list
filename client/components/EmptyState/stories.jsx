import React from 'react';
import { storiesOf } from '@storybook/react';
import EmptyState from '.';

storiesOf('EmptyState', module)
  .add('With Required [props]', () => (
    <EmptyState
      title="Title EmptyState"
      image={<img alt="Image EmptyState" /> /* eslint-disable-line */}
    />
  ))
  .add('With Children', () => (
    <EmptyState
      title="Title EmptyState"
      image={<img alt="Image EmptyState" /> /* eslint-disable-line */}
    >
      <p align="center">Children EmptyState</p>
    </EmptyState>
  ))
  .add('With Description', () => (
    <EmptyState
      title="Title EmptyState"
      description="Description EmptyState"
      image={<img alt="Description EmptyState" /> /* eslint-disable-line */}
    />
  ));
