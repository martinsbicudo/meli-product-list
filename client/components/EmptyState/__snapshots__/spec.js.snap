// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - <EmptyState /> Wrapper Snapshot 1`] = `
<DocumentFragment>
  <div
    class="EmptyState"
  >
    <figure
      class="EmptyStateFigure"
    >
      <img
        alt="TEST_IMAGE"
      />
    </figure>
    <h1
      class="EmptyStateTitle"
    >
      TEST_TITLE
    </h1>
    <p
      class="EmptyStateDescription"
    />
    <span />
  </div>
</DocumentFragment>
`;
